File: GFG/Container With Most Water/Container With Most Water.cpp
long long maxArea(long long A[], int len)
{
    // Your code goes here
    int i = 0;
    int j = len-1;
    
    long long maxArea = 0;
    
    long long base,height, currArea;
    while(i < j)
    {
        base = j-i;
        height = min(A[i], A[j]);
        currArea = base*height;
        
        maxArea = max(maxArea, currArea);
        
        if(A[i] <= A[j]) i++;
        else j--;
    }
    
    return maxArea;
}

File: GFG/Container With Most Water/Container With Most Water.java

class Solve{
    
    long maxArea(int A[], int len){
        // Code Here
        long ans=0;
        int i=0;
        int j=len-1;
        while(i<j){
            ans=Math.max(ans,Math.min(A[i],A[j])*(j-i));
            if(A[i]<A[j]){
                i++;
            }else{
                j--;
            }
        }
        return ans;
    }
    
}
